odoo/addons/base/models/ir_model.py

```diff
@@ -871,8 +871,8 @@ def _prepare_update(self):
             if not uninstalling:
                 field, dep = failed_dependencies[0]
                 raise UserError(_(
-                    "The field '%s' cannot be removed because the field '%s' depends on it.",
-                    field, dep,
+                    "The field '%(field)s' cannot be removed because the field '%(dep)s' depends on it.",
+                    field=field, dep=dep,
```

Comment:
```suggestion
                    "The field '%(field)s' cannot be removed because the field '%(other_field)s' depends on it.",
                    field=field, other_field=dep,
```